### PR TITLE
Wallet object contagion

### PR DESCRIPTION
**Pull request by Kukks**
proof of concept for the spread of wallet object links

image

### CODE REVIEW

File: BTCPayServer/Services/WalletRepository.cs [hunk 1]
```diff
@@ -303,18 +323,59 @@ public async Task EnsureWalletObjectLink(WalletObjectId a, WalletObjectId b, JOb
...
+            if (contagion is null)
+            {
+                await EnsureWalletObjectLink(a, b, data, infectionRate);
+            }
+            if ( contagion.InfectionRate == 0)
+            {
+                return;
+            }
+
+            foreach (WalletObjectData parentNeighbour in contagion.Parent.GetNeighbours())
```
Comment (NicolasDorier) [inline BTCPayServer/Services/WalletRepository.cs:369]: `contagion.GetNeighbours` already returns both parent and child, so you don't need to duplicate this foreach with the one below.

File: BTCPayServer/Controllers/UIWalletsController.cs [hunk 2]
```diff
@@ -578,17 +578,30 @@ private async Task<bool> CanUseHotWallet()
                 var utxos = await _walletProvider.GetWallet(network)
                     .GetUnspentCoins(schemeSettings.AccountDerivation, false, cancellation);
 
-                var walletTransactionsInfoAsync = await this.WalletRepository.GetWalletTransactionsInfo(walletId, utxos.Select(u => u.OutPoint.Hash.ToString()).Distinct().ToArray());
+                var walletTransactionsInfoAsync = await this.WalletRepository.GetWalletTransactionsInfo(walletId, 
+                    utxos.SelectMany(u => new []
+                    {
+                        new ObjectTypeId(WalletObjectData.Types.Tx, u.OutPoint.Hash.ToString()),
+                        new ObjectTypeId(WalletObjectData.Types.Script,  u.ScriptPubKey.ToString()),
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/UIWalletsController.cs:585]: id's should be `.ToHex` not `.ToString`.

**Comment by Jombie69** [comment]
How do I get my money?

**Comment by Clarkv61** [comment]
If anyone has the knowledge to help me withdraw the funds I will pay very well when the funds arrive.

File: BTCPayServer/Services/WalletRepository.cs (hunk 1 above)
Comment (ittybittybit) [reply BTCPayServer/Services/WalletRepository.cs:382]: ok thank you for the heads up Nicolas

**Pull request closed without merging**